tests/misc/test_signed_serializable.py

```diff
@@ -58,3 +58,19 @@ class Unsigned(ssz.Serializable):
     signed = Signed(123, b"\xaa", b"\x00")
     unsigned = Unsigned(123, b"\xaa")
     assert signed.signing_root == unsigned.root
+
+
+def test_eq():
+    class Signed(ssz.SignedSerializable):
+        fields = (
+            ("field1", uint8),
+            ("field2", byte_list),
+            ("signature", byte_list),
+        )
+
+    signed_a = Signed(123, b"\xaa", b"\x00")
+    signed_b = signed_a.copy()
+    assert signed_a == signed_b
+
+    signed_b = signed_b.copy(field1=456)
+    assert signed_a != signed_b
```

Comment:
Maybe add a test that checks that a signed serializable is equal to an unsigned serializable if all fields match.

Reply:
That API is surprising to me.  Seems like equality checks between those two returning `True` might violate principle of least surprise.

Reply:
To clarify: A signed serializable is equal to an unsigned serializable only if they have the same fields *including* the signature. This is what I would expect: The two Python objects represent the same SSZ object (same fields, same data, same root). The only difference is that the `SignedSerializable` has some additional convenience features.

Reply:
@jannikluhn @pipermerriam Thanks for the feedback!

Added more tests: https://github.com/ethereum/py-ssz/pull/64/commits/5e32ef2bf3295a67ca2d4db7969453a3293402ac, and yep, a little surprising that signed serializable `==` to an unsigned serializable.

Should we change it to checking the same `__class__`? i.e.,:
```python
    def __eq__(self, other):
        return self.__class__ == other.__class__ and hash(self) == hash(other)
```

Reply:
I left the previous comment before I saw Jannik's latest comment.

I suppose for the use case, we would use `==` for the fields check and `is` for distinguishing the object.

`SignedSerializable` might be a special case; another case is that do we expect a `PetersburgBlock` object `==` a `ByzantiumBlock` object?

Reply:
so IIUC:

```python
class Thing(Serializable):
    fields = ...

class SignedThing(SignedSerializable):
    fields = ...

thing = Thing(...)
signed_thing = SignedThing(...)  # same field values as `thing` but with signature

# we expect this to be true
assert thing == signed_thing
```

If so, then this is indeed surprising to me and doesn't seem like a good API choice.

Reply:
What would be the less confusing behavior be instead? I can see four options:

1) equal only if class is equal
3) equal if one class is the same or a subclass of the other
2) equal if both are `Serializable` or both are `SignedSerializable`, but not if one is signed and the other isn't
4) equal if both inherit from the same `fields` defining parent

Reply:
I'd say (1). In the use cases, we won't have a `Thing` and a `SignedThing` with the same fields.

Reply:
I think 1 is the safest and easiest to relax the constraint to become 3 or 4 if the use case proves useful.